akka-stream-typed/src/main/scala/akka/stream/typed/scaladsl/ActorMaterializer.scala

```diff
@@ -25,4 +26,19 @@ object ActorMaterializer {
   def apply[T](materializerSettings: Option[ActorMaterializerSettings] = None, namePrefix: Option[String] = None)(implicit actorSystem: ActorSystem[T]): ActorMaterializer =
     akka.stream.ActorMaterializer(materializerSettings, namePrefix)(actorSystem.toUntyped)
 
+  /**
+   * Creates an `ActorMaterializer` which will execute every step of a transformation
+   * pipeline within its own [[akka.actor.Actor]]. The lifecycle of the materialized streams
+   * will be bound to the lifecycle of the provided [[akka.actor.typed.scaladsl.ActorContext]]
+   *
+   * The materializer's [[akka.stream.ActorMaterializerSettings]] will be obtained from the
+   * configuration of the `context`'s underlying [[akka.actor.typed.ActorSystem]].
+   *
+   * The `namePrefix` is used as the first part of the names of the actors running
+   * the processing steps. The default `namePrefix` is `"flow"`. The actor names are built up of
+   * `namePrefix-flowNumber-flowStepNumber-stepName`.
+   */
+  def fromContext[T](ctx: ActorContext[T], materializerSettings: Option[ActorMaterializerSettings] = None, namePrefix: Option[String] = None): ActorMaterializer =
```

Comment:
Not sure I like the name `fromContext`. Is there a problem with having an overloaded `apply`?

Reply:
I can't overload the `apply` method since it has default arguments. Hence the name `fromContext`